### PR TITLE
session: support non-transactional insert statements

### PR DESCRIPTION
**Pull request by you06**
### What problem does this PR solve?

Issue Number: ref #33485
Problem Summary:

### What is changed and how it works?

This PR supports non-transaction insert statements.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [x] Affects user behaviors
- [x] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
Support non-transaction insert statements.
```

### CODE REVIEW

File: session/nontransactional.go [hunk 1]
```diff
@@ -76,6 +76,13 @@ func (j job) String(redacted bool) string {
 
 // HandleNonTransactionalDML is the entry point for a non-transactional DML statement
 func HandleNonTransactionalDML(ctx context.Context, stmt *ast.NonTransactionalDMLStmt, se Session) (sqlexec.RecordSet, error) {
+	// disable staleread for NT transactions because they always write.
+	sessVars := se.GetSessionVars()
+	stale := sessVars.ReadStaleness
+	sessVars.ReadStaleness = 0
+	defer func() {
+		sessVars.ReadStaleness = stale
```
Comment (ekexium) [inline session/nontransactional.go:84]: Staleness(and select limit) was handled around line 410. Shall we move them here?

**Review by ekexium — COMMENTED** [review]
Shall we add a constraint that the source table cannot be the dest table?

File: session/nontransactional.go (hunk 1 above)
Comment (you06) [reply session/nontransactional.go:84]: I think so, before disabling stale read, it went wrong when preprocessing in L86.

**Review by ekexium — APPROVED** [review]
> Shall we add a constraint that the source table cannot be the dest table?

We can discuss it later

**Comment by you06** [comment]
I just added the test cases for `insert on duplicate key update` statements.

**Comment by you06** [comment]
> Shall we add a constraint that the source table cannot be the dest table?

Disable this use case by now.

**Review by cfzjywxk — APPROVED** [review]

**Pull request merged**